specifyweb/frontend/js_src/lib/components/Header/Notifications.tsx

```diff
@@ -133,13 +133,14 @@ export function Notifications({
           });
           lastFetchedTimestamp = startFetchTimestamp;
           // Stop updating if tab is hidden
-          return (timeout =
+          timeout =
             document.visibilityState === 'hidden'
               ? undefined
               : globalThis.setTimeout(
                   () => doFetch(lastFetchedTimestamp),
                   pullInterval
-                ));
+                );
+          return timeout;
```

Comment:
(optional) you don't actually need to return anything here as it's not being read by anyone?

Reply:
I had to add `return timeout` because of the error 'Not all code paths return a value.' on line 102 in the .then
Is there another solution for it?

Reply:
yes,
see the eslint doc for that page  -https://eslint.org/docs/latest/rules/consistent-return

to fix, replace `return undefined;` with `return` earlier in this function - then you won't need `return undefined`